sql/core/src/main/scala/org/apache/spark/sql/execution/python/ExtractPythonUDFs.scala

```diff
@@ -137,15 +138,18 @@ object ExtractPythonUDFs extends Rule[SparkPlan] with PredicateHelper {
           udf.references.subsetOf(child.outputSet)
         }
         if (validUdfs.nonEmpty) {
-          if (validUdfs.exists(_.pythonUdfType == PythonUdfType.PANDAS_GROUPED_UDF)) {
-            throw new IllegalArgumentException("Can not use grouped vectorized UDFs")
-          }
+          require(validUdfs.forall(udf =>
+            udf.evalType == PythonEvalType.SQL_BATCHED_UDF ||
+            udf.evalType == PythonEvalType.PANDAS_SCALAR_UDF
+          ), "Can only extract scalar vectorized udf or sql batch udf")
```

Comment:
```require(validUdfs.forall(_.evalType != PythonEvalType.PANDAS_GROUP_MAP_UDF))```?

Reply:
I think "equal" comparison is better than "not equal", because we might add new types.